nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -311,9 +311,9 @@ let
             else defaultListen;
 
         listenString = { addr, port, ssl, extraParameters ? [], ... }:
-          (if ssl && vhost.http3 then "
-          # UDP listener for **QUIC+HTTP/3
-          listen ${addr}:${toString port} http3 "
+          # UDP listener for QUIC transport protocol.
+          (if ssl && vhost.quic then "
+          listen ${addr}:${toString port} quic "
           + optionalString vhost.default "default_server "
           + optionalString vhost.reuseport "reuseport "
           + optionalString (extraParameters != []) (concatStringsSep " " extraParameters)
```

Comment:
So, taking the commit in hg, I see:
```
Now "ssl", "proxy_protocol" and "http2" are not allowed with "quic" in "listen"
directive.  Previously, only "ssl" was not allowed.
```

So, you could go like this:

```
let incompatibleParameters = [ "ssl" "proxy_protocol" "http2" ];
     isCompatibleParameter = param: !(any (p: p == param) incompatibleParameters);
in filter isCompatibleParameter extraParameters
```

(I let you adapt it to the exact code.)

Therefore, it's impossible to get a broken configuration especially if we generate the listen string ourselves.

Also, we probably want to have assertions errors for cases where the user explicitly set a "proxy_protocol"-only listener or "http2"-only listener, but that's probably out of scope for this PR as it needs to move those parameters in the signature and remove them from `extraParameters`.

Reply:
(At least, the filter is needed because we could generate impossible NGINX configurations.)

Reply:
Hmm, could probably delete the line. 
```
+ optionalString (extraParameters != []) (concatStringsSep " " extraParameters)
```

Reply:
I mean, I don't know what are other parameters compatible with QUIC, but we should probably keep it as it would constrain uselessly the QUIC configuration.
But we should filter out all **known incompatible** parameters.

Reply:
Resolved.